### ISSUE TITLE
Air-by-Date isn't totally fixed. It only imports SOME episodes

### ISSUE THREAD

**Issue by PrSinister** [gboudreau/XBMCnfoTVImporter.bundle#67]
The Air-by-date fix isn't totally working. It only imports the 1st season and only imports episodes 1 to 10, then skips to 20 to 29 then 40 and up.

I have attached more metadata and will list what is being imported: 

CastingCouch-X.com:
     Season 2012: Episodes 8 and 20-26 are imported. Rest aren't. It also thinks that all files with the same date are parts of episode 8
     Season 2013: No episodes imported
     Season 2014: No episodes imported

CzechBitch.com:
     Season 2013: Episodes 8 and 20-23 are imported. Rest aren't. It also thinks that all files with the same date are parts of episode 8
     Season 2014: No episodes imported
	 
CzechCouples.com
	 Season 2014: Only Episodes 10 and 11 are imported
	 
CzechGardenParty.com
     Season 2014: Only the last episode is imported with all files being parts of the episode. These      nfos don't have any <episode /> tag (This works fine in XBMC/Kodi)
	 
CzechParties.com
     Season 2011: Episodes 1-9 and 20-24 are imported. Rest aren't.
	 
CzechStreets.com
     Season 2011: Episodes 1-9, 20-29 and 40-49 are imported. Rest aren't.
	 Season 2012: No episodes imported
	 Season 2013: No episodes imported
	 Season 2014: No episodes imported
	 
Her-Sweet-Hand.com
     Season 2004: Episode 1 is imported (only episode in season)
	 Seasons 2005-2014: No episodes imported
	 
TrickySpa.com
     Season 2014: Episodes 1-9 and 20-29 are imported. Rest aren't.
	 
I have included screenshots from Plex and XBMC so you can see that this works perfectly in XBMC with the current nfos and Plex is not quite working. For the XBMC screenshots, they are of the whole page so you may need to zoom in to see properly.

The Metadata is available in the rar below:
	 
https://dl.dropboxusercontent.com/u/824824/XBMC%20Air-by-date%20v2.rar

**Comment by SlrG** [comment]
Thanks for posting the metadata. I'll have a look at it. Might take a little longer this time however, as the christmas holidays are over.

**Comment by SlrG** [comment]
The problem is quite difficult to grasp. It seems to be rooted within plex itself or maybe in the communication between the agent and plex. I'll have to research a little further if there are things I could change in the agent to solve it. At the moment I don't think I can really work it out.

I started with your CzechCouples.com sample. For me all episodes are imported. There are no error messages generated by the agent. In the logs all episodes are fine. Afterwards plex groups all episodes with the same date together so in the end there are only 4 episodes visible. I can't promise you I can prevent this grouping in the agent.

If anybody knows a way, please feel free to chime in. :)

**Comment by PrSinister** [comment]
Yeah looks like Plex is thinking the ones with the same dates are all parts of the same file...

At this point, if you can just get it to import the ones that are not the same dates as individual files, that would be great. Usually, the 1st batch of videos released has the same dates and then they release one new episode a week.

If you look at the Daily Show, it doesn't import the metadata but they don't even have the same dates...

**Comment by PrSinister** [comment]
Also, for me, Czech Couples only imports metadata for Couples 10 and 11 whlie all others are scanned in as single episodes but only using a Plex generated thumb, the episode title is the date and no description

**Comment by SlrG** [comment]
I'm still trying to solve this problem. Sadly I have no solution at the moment. Latest experiment was to use the Extended Personal Media Scanner (https://forums.plex.tv/index.php/topic/88982-rel-extended-personal-media-scanner/) in conjunction with the xbmc agent. Sadly that does not work also. 

But maybe as a temporary fix, you could use that scanner and the corresponding agent (https://forums.plex.tv/index.php/topic/83440-rel-extended-personal-media-shows-agent/) for your shows?

**Comment by PrSinister** [comment]
Thank you so much for still working on this. If only the problem was only with files with the same date in the filename then i wouldn't mind so much. But the fact that it doesn't even most of the rest is baffling.

I mean i don't understand why it would work this way when clearly there is an NFO available every single file. You would think it would mean we WANT you to look at each file individually :)

That scanner you pointed to doesn't read nfos and sadly that's the whole point. My NFOs have actual useful metadata such as Actors, Titles and Plot info... I guess i will have to just hope that Kodi will get their asses in gear and allow us to access our libraries remotely. In the meantime, i will switch to MediaBrowser for the custom NFos... Their XBMC/Kodi importer doesn't have that problem.

Again, thanks so much for all your hard work on this!

**Comment by SlrG** [comment]
I expected that the missing nfo part wouldn't be to your liking. :) I would hate it too. I'm glad you have found an alternative in using MediaBrowser. I'll leave the issue open and try to solve it, if I get more info or ideas what could be the reason. When there are news I'll report them here.
